spec/models/topic_spec.rb

```diff
@@ -32,6 +32,16 @@
       @private_topic = Topic.create!(name: RandomData.random_sentence, description: RandomData.random_paragraph, public: false)
     end
 
+    describe "publicly_viewable" do
+      it "returns all topics that are publicly viewable"
+      expect(Topic.publicly_viewable).to eq([@public_topic])
+    end
+
+    describe "privately_viewable" do
+      it "returns all topics that are privately viewable"
+      expect(Topic.privately_viewable).to eq([@private_topic])
```

Comment:
You might need to make it 

```ruby
expect(Topic.privately_viewable.to_a).to eq([@private_topic])
```

...because a scope returns a relation, not an array. But we can easily convert it to an array of results with `#to_a`.